code/obj/item/go_board.dm

```diff
@@ -0,0 +1,196 @@
+				var color = href_list["color"]
+
+				if(!(src.GetOverlayImage("[href_list["position"]]")))
+					var/image/piecedisplay = new /image('icons/obj/items/items.dmi',"go-render-stone-[color]")
+					piecedisplay.pixel_x = (-offsetx)
+					piecedisplay.pixel_y = (-offsety)
+
+					src.UpdateOverlays(piecedisplay,"[href_list["position"]]")
+
+				for(var/mob/living/carbon/human/u in src.openwindows)
+					if(u == usr)
+						continue
+					if(u.client && !(u in range(u.client.view,src)))
+						u << browse(null, "window=go")
+						break
+					u << output(list2params(list("null","null","[color]","null","secondary","[href_list["position"]]")),"go.browser:createpiece")
+				return
+			if("remove")
+				if(istype(usr,/mob/living/carbon/human) && (usr in range(1,src)))
+					var/mob/living/carbon/human/user = usr
+					src.ClearSpecificOverlays("[href_list["position"]]")
+
+					if(href_list["color"]=="black")
+						user.put_in_hand_or_drop(new /obj/item/gostoneb)
+					else
+						user.put_in_hand_or_drop(new /obj/item/gostonew)
+
+					for(var/i=1,i<=src.piecelist.len,i++)
+						if("[src.piecelist[i]["position"]]"==href_list["position"])
+							src.piecelist[i] = null
+							src.piecelist.Remove(null)
+					for(var/mob/living/carbon/human/u in src.openwindows)
+						if(u == user)
+							continue
+						if(u.client && !(u in range(u.client.view,src)))
+							u << browse(null, "window=go")
+							break
+						u << output("[href_list["position"]]","go.browser:removepiece")
+					return
+				else if(usr.client && !(usr in range(usr.client.view,src)))
+					usr << browse(null, "window=go")
+					return
+
+	MouseDrop(mob/user as mob)
+		if((istype(user,/mob/living/carbon/human))&&(!user.stat)&&!(src in user.contents))
+			user.put_in_hand_or_drop(src)
+
+	disposing()
+		for(var/mob/living/carbon/human/user in src.openwindows)
+			user << browse(null, "window=go")
+		..()
+
+/obj/item/gobowlb
+	name = "bowl of black stones"
+	desc = "a bowl of little black stones."
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "go-bowl-b"
+
+	attack_hand(mob/user as mob)
+		user.put_in_hand_or_drop(new /obj/item/gostoneb)
+
+	attackby(obj/item/weapon as obj,mob/user as mob)
+		if((istype(weapon, /obj/item/gostoneb)) || (istype(weapon, /obj/item/gostonew)))
+			user.u_equip(weapon)
+			qdel(weapon)
+
+	MouseDrop(mob/user as mob)
+		if((istype(user,/mob/living/carbon/human))&&(!user.stat)&&!(src in user.contents))
+			user.put_in_hand_or_drop(src)
+
+	MouseDrop_T(atom/movable/O as mob|obj, mob/user as mob)
+		var/total
+		var/color
+		if(istype(O,/obj/item/gostoneb))
+			color = "black"
+			for (var/obj/item/gostoneb/s in range(1, user))
+				qdel(s)
+				total++
+				sleep(2)
+		else if(istype(O,/obj/item/gostonew))
+			color = "white"
+			for (var/obj/item/gostonew/s in range(1, user))
+				qdel(s)
+				total++
+				sleep(2)
+		src.visible_message("<span><b>[usr]</b> adds [total] [color] stones to the bowl!</span>")
+
+/obj/item/gobowlw
```

Comment:
I'd make a common parent `/obj/item/gobowl` with subtypes for white and for black, it'd cut down on duplicate code considerably and shouldn't be too much effort (really just `attack_hand` code will need to depend on some var on the bowl that determines the type).

Reply:
Done in next commit :D